### PR TITLE
feat: run docker container as non root user

### LINKED ISSUE #6505: Run docker container as non root user
Currently, the app and celery run as root in docker containers 

They should be run as normal users

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #6505 

#### Short description of what this resolves:

feat: run docker container as non root use

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal - Check this

**Comment by iamareebjamal** [comment]
Did you test this?

**Comment by codedsun** [comment]
@iamareebjamal Yes

**Comment by codedsun** [comment]
Also for the celery container, I was getting some error as permission denied. Please look into this. But the instance was running and the celery was working fine.

image


image

**Comment by iamareebjamal** [comment]
You need to change the permission of workdir for open_event_user to be able to write in it

**Comment by codedsun** [comment]
@iamareebjamal How do i do this?

**Comment by iamareebjamal** [comment]
search

**Comment by codedsun** [comment]
@iamareebjamal - Working fine now

```
opev-celery | [2019-12-28 18:20:33,904: INFO/MainProcess] Connected to redis://redis:6379/0
opev-celery | [2019-12-28 18:20:33,937: INFO/MainProcess] mingle: searching for neighbors
opev-celery | [2019-12-28 18:20:35,011: INFO/MainProcess] mingle: all alone
opev-celery | [2019-12-28 18:20:35,109: INFO/MainProcess] celery@56d327afcd2b ready.
opev-celery | [2019-12-28 18:20:37,443: INFO/Beat] beat: Starting...
opev-web    | INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
opev-web    | INFO  [alembic.runtime.migration] Will assume transactional DDL.
opev-web    | [LOG] Starting gunicorn on port 8080
opev-web    | [2019-12-28 18:20:45 +0000] [39] [INFO] Starting gunicorn 20.0.4
opev-web    | [2019-12-28 18:20:45 +0000] [39] [INFO] Listening at: http://0.0.0.0:8080 (39)
opev-web    | [2019-12-28 18:20:45 +0000] [39] [INFO] Using worker: sync
opev-web    | [2019-12-28 18:20:45 +0000] [51] [INFO] Booting worker with pid: 51
opev-web    | [2019-12-28 18:20:45 +0000] [52] [INFO] Booting worker with pid: 52
opev-web    | [2019-12-28 18:20:45 +0000] [53] [INFO] Booting worker with pid: 53
opev-web    | [2019-12-28 18:20:45 +0000] [54] [INFO] Booting worker with pid: 54
```

File: Dockerfile [hunk 1]
```diff
@@ -25,5 +25,11 @@ RUN apk --no-cache add postgresql-libs ca-certificates libxslt jpeg zlib file li
 WORKDIR /data/app
 ADD . .
 
+RUN adduser -S open_event_user
+RUN chown -R open_event_user /data/app
+RUN chmod 755 /data/app
```
Comment (iamareebjamal) [inline Dockerfile]: Why chmod?

**Comment by iamareebjamal** [comment]
Remove FORCE ROOT in docker compose

**Comment by codedsun** [comment]
@iamareebjamal Check now!

**Comment by iamareebjamal** [comment]
tested?

**Comment by codedsun** [comment]
@iamareebjamal Yes. 

Log

```
opev-postgres | 2019-12-28 18:33:18.143 UTC [20] LOG:  database system was interrupted; last known up at 2019-12-28 18:20:22 UTC
opev-web    | [LOG] Database Up
opev-web    | [LOG] Preparing database
opev-postgres | 2019-12-28 18:33:18.607 UTC [20] LOG:  database system was not properly shut down; automatic recovery in progress
opev-postgres | 2019-12-28 18:33:18.612 UTC [20] LOG:  redo starts at 0/18F7038
opev-postgres | 2019-12-28 18:33:18.612 UTC [20] LOG:  invalid record length at 0/18F7070: wanted 24, got 0
opev-postgres | 2019-12-28 18:33:18.612 UTC [20] LOG:  redo done at 0/18F7038
opev-postgres | 2019-12-28 18:33:18.673 UTC [1] LOG:  database system is ready to accept connections
opev-postgres | 2019-12-28 18:33:19.490 UTC [27] LOG:  incomplete startup packet
opev-celery | [LOG] Deploying celery
opev-celery | [LOG] Using database: postgresql://open_event_user:opev_pass@postgres:5432/open_event
opev-celery | [LOG] Using redis: redis://redis:6379/0
opev-celery | [LOG] Starting celery worker
opev-web    | [LOG] Existing tables:
opev-web    | [LOG] ticket_fees,settings,orders,event_topics,feedback,email_notifications,speakers_sessions,roles_panels,panel_permissions,services,roles,ticket_tagging,user_permissions,users_events_roles,speaker,access_codes_tickets,discount_codes_tickets,events,access_codes,activities,ticket_holders,mails,image_sizes,event_copyrights,events_orga,stripe_authorizations,message_settings,event_locations,custom_form_options,sessions_speakers_links,session_types,event_types,sponsors,tracks,social_links,modules,microlocations,role_invites,faq_types,user_emails,pages,sessions_version,user_token_blacklist_time,user_favourite_events,events_version,speakers_sessions_version,custom_placeholders,export_jobs,orders_tickets,import_jobs,notification_actions,user_system_role,tax,speakers_calls,event_invoices,users,discount_codes,ticket_tag,transaction,tickets,event_sub_topics,custom_forms,alembic_version,notifications,faq,permissions,custom_sys_roles,sessions
opev-web    | [LOG] Tables already exist. Skipping data population & creation.
opev-web    | [LOG] Running migrations
opev-celery | [2019-12-28 18:33:30,239: INFO/MainProcess] Connected to redis://redis:6379/0
opev-celery | [2019-12-28 18:33:30,289: INFO/MainProcess] mingle: searching for neighbors
opev-celery | [2019-12-28 18:33:31,377: INFO/MainProcess] mingle: all alone
opev-celery | [2019-12-28 18:33:31,466: INFO/MainProcess] celery@e7b62699c903 ready.
opev-celery | [2019-12-28 18:33:34,190: INFO/Beat] beat: Starting...
opev-web    | INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
opev-web    | INFO  [alembic.runtime.migration] Will assume transactional DDL.
opev-web    | [LOG] Starting gunicorn on port 8080
opev-web    | [2019-12-28 18:33:44 +0000] [39] [INFO] Starting gunicorn 20.0.4
opev-web    | [2019-12-28 18:33:44 +0000] [39] [INFO] Listening at: http://0.0.0.0:8080 (39)
opev-web    | [2019-12-28 18:33:44 +0000] [39] [INFO] Using worker: sync
opev-web    | [2019-12-28 18:33:44 +0000] [51] [INFO] Booting worker with pid: 51
opev-web    | [2019-12-28 18:33:44 +0000] [52] [INFO] Booting worker with pid: 52
opev-web    | [2019-12-28 18:33:44 +0000] [53] [INFO] Booting worker with pid: 53
opev-web    | [2019-12-28 18:33:44 +0000] [54] [INFO] Booting worker with pid: 54

```

**Comment by codedsun** [comment]
@iamareebjamal - Thankyou!

**Comment by iamareebjamal** [comment]
Image upload is not working. Permission Denied @codedsun

**Comment by iamareebjamal** [comment]
Reverting for now

**Comment by codedsun** [comment]
@iamareebjamal Because that goes to the `static` folder imo and we have only given permission of the directory `/data/app`

**Comment by iamareebjamal** [comment]
Yes

**Comment by codedsun** [comment]
@iamareebjamal Now the image upload is working, I have given the permission. But the celery task is failing and  says `Connection Refused`

Log:

```
opev-celery | [2019-12-29 06:56:21,345: INFO/MainProcess] Received task: resize.user.images[2c151344-4a18-43d9-8aab-142356519e1e]  
opev-celery | [2019-12-29 06:56:21,460: INFO/ForkPoolWorker-2] User image resizing tasks started http://0.0.0.0:8080/static/media/temp/images/8069f903-efdb-416d-a56b-e396e7da1968/VW85WXRCSm/04a217d6-a5dd-464c-afd3-63d0c5ed7a53.jpeg
opev-celery | [2019-12-29 06:56:21,584: ERROR/ForkPoolWorker-2] Task resize.user.images[2c151344-4a18-43d9-8aab-142356519e1e] raised unexpected: ConnectionError(MaxRetryError("HTTPConnectionPool(host='0.0.0.0', port=8080): Max retries exceeded with url: /static/media/temp/images/8069f903-efdb-416d-a56b-e396e7da1968/VW85WXRCSm/04a217d6-a5dd-464c-afd3-63d0c5ed7a53.jpeg (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f75f8b057d0>: Failed to establish a new connection: [Errno 111] Connection refused'))"))
opev-celery | Traceback (most recent call last):
opev-celery |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 157, in _new_conn
opev-celery |     (self._dns_host, self.port), self.timeout, **extra_kw
opev-celery |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 84, in create_connection
opev-celery |     raise err
opev-celery |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 74, in create_connection
opev-celery |     sock.connect(sa)
opev-celery | ConnectionRefusedError: [Errno 111] Connection refused
```

Can you guide me what's wrong?

**Comment by iamareebjamal** [comment]
URL is wrong `http://0.0.0.0:8080/static/media/temp/images/8069f903-efdb-416d-a56b-e396e7da1968/VW85WXRCSm/04a217d6-a5dd-464c-afd3-63d0c5ed7a53.jpeg`

It can't connect to 0.0.0.0:8080

**Comment by codedsun** [comment]
@iamareebjamal - Where can it be corrected?

**Comment by iamareebjamal** [comment]
The URL file system need to be refactored. It won't work locally for now. Create a WIP PR so I can test on staging if it works or not

**Pull request merged**